twikiget/cli.py

```diff
@@ -55,3 +66,99 @@ def wget_warc(url, wget_options, out_warc_file, directory_prefix):
                       directory_prefix=directory_prefix,
                       include=include)
               )
+
+
+@twikiget.command('list')
+@click.option(
+    '--json',
+    'output_format',
+    flag_value='json',
+    default=None,
+    help='Get output in JSON format.')
+@click.option(
+    '--content-type',
+    'filter_content_type',
+    default='',
+    help='Filter files in an archive by content_type. '
+         'It can be either full version or just a begging of type name')
+@click.argument('warc_file')
+def list(warc_file, output_format, filter_content_type):
+    """ List files in warc archive.
+
+    The list can be filtered by the HTTP Content Type,
+     and exported as json if needed.
+
+    Examples: \n
+    \t $ twikiget list ExampleTwiki.warc \n
+    \t ...
+
+    \t $ twikiget list ExampleTwiki.warc --content-type=text/html\n
+    \t ...
+
+    \t $ twikiget list ExampleTwiki.warc --content-type=text\n
+    \t ...
+
+    \t $ twikiget list ExampleTwiki.warc --json
+    \t ...
+
+    """
+    tablib_data = tablib.Dataset()
+    tablib_data.headers = ['uri', 'title', 'content_type']
+    with open(warc_file, 'rb') as fh:
+        for r in ArchiveIterator(fh):
+            if (r.rec_type == 'response' and
+                    r.http_headers.get_statuscode().startswith('2')):
+                content_type = r.http_headers.get_header('content-type')
+                if content_type.lower().startswith(filter_content_type):
+                    title = None
+                    uri = r.rec_headers.get_header('warc-target-uri')
+                    if content_type.lower().startswith('text/html'):
+                        payload = r.content_stream().read()
+                        title = BeautifulSoup(payload, 'html.parser').title
+                    tablib_data.append(
+                        [str(uri), str(title), str(content_type)])
+
+        if output_format:
+            click.echo(tablib_data.export(output_format))
+        else:
+            table = PrettyTable()
+            table.field_names = ['uri', 'title', 'content_type']
+            for row in tablib_data:
+                table.add_row(row)
+            click.echo(table)
+
+
+@twikiget.command()
+@click.argument('warc-file')
+@click.argument('file-uri')
+def view(warc_file, file_uri):
+    """ Get content of one of the files in warc archive.
```

Comment:
- "View raw content..." instead of "Get".

- Note that WARC is used in uppercase usually, please change here and everywhere.

- Also, while you are at it, please shorten the help text for "archive" so that the output would look nice:
```
$ twikiget --help
Usage: twikiget [OPTIONS] COMMAND [ARGS]...

Options:
  --help  Show this message and exit.

Commands:
  archive  Downloading a page with attachments to a catalog and into a warc...
  list     List files in warc archive.
  view     Get content of one of the files in warc archive.
```